### ISSUE TITLE
Representation descriptive metadata indexing and searching

### ISSUE THREAD

**Issue by beepsoft** [keeps/roda#1097]
Hi,

Representations may have descriptive metadata similar to AIPs, but I found that these metadata values are not indexed at all, and hence cannot be searched for. Is there a reason for this? Without indexing the metadata the Representation search function seem a little bit useless: you can only search by identifier, type and file type/size.

Thanks for any clarifications!

**Comment by luis100** [comment]
hi @beepsoft, they are indexed in the same way AIPs are.

Check [this comment](https://github.com/keeps/roda/issues/1062#issuecomment-343954018) on how to generically enable advanced search and [this comment]( https://github.com/keeps/roda/issues/1075#issuecomment-345861652) on how to extend it to representations.

We know that documentation about this is a bit scarce, but we can help if you have more specific questions.

**Comment by luis100** [comment]
I just noted that you might encounter here a problem if you try to use the default schemas like EAD or Dublin Core to add desc. metadata to the representation level because the configuration of the indexing of these metadata schemas will use index fields that only exist on the Intellectual Entity level, specifically title, initial and final date. This will make the indexing of the representation metadata to fail.

To enable this, you need to create a specific metadata schema type and version with a configuration for indexing that will be appropriate for representations, i.e. only using the index extension fields *_txt, *_ss, *_dd, etc. For example, the index config should not create the index field "title" but instead should create the index field "title_txt".

It is indeed a bit cumbersome, we will work on making this easier in the future.

**Comment by beepsoft** [comment]
Hi,

thanks for your answer!

I could work around that and now I have indexing for Representations. 

But I had to to more than just updating the indexing schema.xml. For example, IndexModelObserver#descriptiveMetadataCreated and IndexModelObserver#descriptiveMetadataUpdated only starts indexing the descriptive metadata if it belongs to an AIP (descriptiveMetadata.isFromAIP()). 

These are all the changes I had to apply to make it work for any descriptive metadata:

[representation-indexing.patch.txt](https://github.com/keeps/roda/files/1513830/representation-indexing.patch.txt)
